lib/board.rb

```diff
@@ -0,0 +1,189 @@
+class Board
+  WIN_COMBINATIONS=[[0,1,2],[3,4,5],[6,7,8],[0,4,8],[2,4,6],[1,4,7],[0,3,6],[2,5,8]] #defines all of the possible combinations to win
+  attr_accessor :cells
+  def reset! #resets the @cells which empties the board
+    @cells=[" "," "," "," "," "," "," "," "," "]
+  end
+    
+  def cells #is the board 
+    @cells
+  end
+   
+  def initialize # resets and creates a new board for a new game
+    self.reset!
+  end
+   
+  def display #displays the board currently in cells
+    print " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} \n"
+    print "-----------\n"
+    print " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} \n"
+    print "-----------\n"
+    print " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+   
+  def position(input) #returns the contents of the selected cell
+    return @cells[input.to_i-1]
+  end
+   
+  def full? #determines if the board is full
+    !@cells.include?(" ") 
+  end
+   
+  def turn_count #returns the amount of turns in the game
+    return @cells.count {|cell| cell!=" "}
+  end
+   
+  def taken?(input) #determines of the input cell (changed) is already taken or not
+    input=input.to_i-1
+    if @cells[input]!=" "
+      return true
+    else
+      return false
+    end
+  end
+   
+  def valid_move?(input) #determines if a given number is a valid number and if the position is taken
+    input.split.all? {|local_input| local_input.to_i<10 && local_input.to_i>0} #turn the input into an array so that i can compare with a number, was getting nomethod error "<" previously
+    if taken?(input) #determines if the input is not taken
+      return false #if the input is taken, then it is not a valid move
+    else
+      return true #if the input is not taken, then it could be a valid move
+    end
+  end
```

Comment:
Reverse (or negate) the boolean with the `!` (bang) operator.

`!taken?(input)`